### PR TITLE
{Tests} Deallocate test objects at the end of tests.

### LINKED ISSUE #5395: {Tests} Nil all properties in tearDown
All properties created in `-setUp` should be deallocated in `-tearDown`.

https://qualitycoding.org/xctestcase-teardown/

### PR DESCRIPTION
**Pull request by romoore**
Deallocating many of the properties created for each unit test. Since all
XCTestCase instances survive until the end of the test suite, any properties
created for a test will be preserved as well. This frees up around 3.7 MB
(~6%) of memory from the entire test suite, as measured on my desktop.

|Before|After|
|----|----|
|test-objcs-before|test-objcs-after|

See also: https://qualitycoding.org/xctestcase-teardown/

Closes #5395

### CODE REVIEW

**Comment by romoore** [comment]
~Note that the Dialogs-related objects appear to be caused by a retain cycle #5396~

It looks like it may just be late deallocation in the tests.

File: components/FlexibleHeader/tests/unit/FlexibleHeaderNoTrackingScrollViewTests.swift [hunk 1]
```diff
@@ -1,26 +0,0 @@
...
-
-// Tests confirming contract with no tracking scroll view
-class FlexibleHeaderNoTrackingScrollViewTests: FlexibleHeaderTrackingScrollViewTests {
-
-  override func setUp() {
-    super.setUp()
-
-    view.trackingScrollView = nil
-  }
-}
```
Comment (codeman7) [inline components/FlexibleHeader/tests/unit/FlexibleHeaderNoTrackingScrollViewTests.swift:26]: Just want to confirm this file isn't being imported anywhere right?

**Review by codeman7 — APPROVED** [review]
_Tip:_ Consistency with 
```
property = nil;

[super teardown];
``` 
or 
```
property = nil;
[super teardown];
```
I believe the former is the preferred style but whatever you choose please just be consistent.

File: components/FlexibleHeader/tests/unit/FlexibleHeaderNoTrackingScrollViewTests.swift (hunk 1 above)
Comment (romoore) [reply components/FlexibleHeader/tests/unit/FlexibleHeaderNoTrackingScrollViewTests.swift:26]: It didn't cause any problems to delete it and I can't find any mentions of it in the code base.

**Comment by romoore** [comment]
@codeman7 I agree with you, but in some cases I was maintaining within-file consistency. I believe keeping files internally consistent is more important (according to the style guide) than improving portions of them incrementally.  On more than one occasion, I've been told not to deviate from the file's style even if it didn't align with the project's style.  I believe we are only supposed to improve a file's style if we do it holistically across the file at once.

**Pull request merged**